File: src/client/java/fr/hardel/asset_editor/client/javafx/components/ui/Card.java
package fr.hardel.asset_editor.client.javafx.components.ui;

import fr.hardel.asset_editor.client.javafx.VoxelColors;
import fr.hardel.asset_editor.client.javafx.VoxelFonts;
import javafx.beans.property.BooleanProperty;
import javafx.beans.property.SimpleBooleanProperty;
import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.control.Label;
import javafx.scene.layout.StackPane;
import javafx.scene.layout.VBox;
import javafx.scene.paint.Color;
import net.minecraft.client.resources.language.I18n;
import net.minecraft.resources.Identifier;

/**
 * Slot card: p-4 card with image (bottom) and title/desc (top).
 * Active:  bg-zinc-950/50 + zinc-700 border ring.
 * Locked:  opacity-50 + zinc-700 border + lock icon + lock text.
 * Clicking toggles active state (local BooleanProperty).
 */
public final class Card extends SimpleCard {

    private static final Identifier CHECK = Identifier.fromNamespaceAndPath("asset_editor", "icons/check.svg");
    private static final Identifier LOCK  = Identifier.fromNamespaceAndPath("asset_editor", "icons/tools/lock.svg");

    private final BooleanProperty active = new SimpleBooleanProperty(false);
    private final boolean locked;

    public Card(Identifier imageId, String titleKey, String descKey, boolean initialActive, boolean locked, String lockKey) {
        super(new Insets(16));
        setMaxWidth(Double.MAX_VALUE);
        this.locked = locked;

        Label title = new Label(I18n.get(titleKey));
        title.setFont(VoxelFonts.of(VoxelFonts.Variant.SEMI_BOLD, 16));
        title.setTextFill(VoxelColors.ZINC_100);

        VBox textBlock = new VBox(4, title);
        textBlock.setAlignment(Pos.TOP_LEFT);
        textBlock.setMaxWidth(Double.MAX_VALUE);
        if (descKey != null) {
            Label desc = new Label(I18n.get(descKey));
            desc.setFont(VoxelFonts.of(VoxelFonts.Variant.REGULAR, 13));
            desc.setTextFill(VoxelColors.ZINC_400);
            desc.setWrapText(true);
            textBlock.getChildren().add(desc);
        }

        ResourceImageIcon image = new ResourceImageIcon(imageId, 64);
        StackPane imageWrap = new StackPane(image);
        imageWrap.setAlignment(Pos.CENTER);
        imageWrap.setMinSize(64, 64);
        imageWrap.setPrefSize(64, 64);
        imageWrap.setMaxSize(64, 64);

        VBox inner = new VBox(textBlock, imageWrap);
        inner.setAlignment(Pos.TOP_CENTER);
        VBox.setMargin(textBlock, new Insets(0, 0, 32, 0));
        VBox.setMargin(imageWrap, new Insets(0, 0, 32, 0));

        contentBox.getChildren().add(inner);

        buildOverlays(lockKey);
        applyActiveState(initialActive);
        active.set(initialActive);

        if (locked) {
            setOpacity(0.5);
        } else {
            setOnMouseClicked(e -> active.set(!active.get()));
            active.addListener((obs, o, v) -> applyActiveState(v));
        }
    }

    public Card(Identifier imageId, String titleKey) {
        this(imageId, titleKey, null, false, false, null);
    }

    public Card(Identifier imageId, String titleKey, boolean initialActive) {
        this(imageId, titleKey, null, initialActive, false, null);
    }

    private void buildOverlays(String lockKey) {
        if (locked) {
            SvgIcon lockIcon = new SvgIcon(LOCK, 24, Color.WHITE);
            StackPane.setAlignment(lockIcon, Pos.TOP_RIGHT);
            StackPane.setMargin(lockIcon, new Insets(16));
            lockIcon.setMouseTransparent(true);
            visualCard.getChildren().add(lockIcon);

            if (lockKey != null) {
                Label lockText = new Label(I18n.get(lockKey));
                lockText.setFont(VoxelFonts.of(VoxelFonts.Variant.LIGHT, 11));
                lockText.setTextFill(VoxelColors.ZINC_400);
                lockText.setMouseTransparent(true);
                StackPane.setAlignment(lockText, Pos.BOTTOM_RIGHT);
                StackPane.setMargin(lockText, new Insets(16));
                visualCard.getChildren().add(lockText);
            }
        } else {
            SvgIcon checkIcon = new SvgIcon(CHECK, 24, Color.WHITE);
            checkIcon.setMouseTransparent(true);
            StackPane.setAlignment(checkIcon, Pos.TOP_RIGHT);
            StackPane.setMargin(checkIcon, new Insets(16));
            checkIcon.setVisible(false);
            visualCard.getChildren().add(checkIcon);

            active.addListener((obs, o, v) -> checkIcon.setVisible(v));
        }
    }

    private void applyActiveState(boolean on) {
        if (on) {
            visualCard.getStyleClass().add("ui-tool-slot-active");
        } else {
            visualCard.getStyleClass().remove("ui-tool-slot-active");
        }
    }

    public BooleanProperty activeProperty() { return active; }
    public boolean isActive() { return active.get(); }
    public void setActive(boolean v) { active.set(v); }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/components/ui/Card.kt
package fr.hardel.asset_editor.client.compose.components.ui

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import fr.hardel.asset_editor.AssetEditor
import fr.hardel.asset_editor.client.compose.StudioColors
import fr.hardel.asset_editor.client.compose.StudioTypography
import net.minecraft.resources.Identifier

private val CHECK_ICON = Identifier.fromNamespaceAndPath(AssetEditor.MOD_ID, "icons/check.svg")
private val LOCK_ICON = Identifier.fromNamespaceAndPath(AssetEditor.MOD_ID, "icons/tools/lock.svg")

@Composable
fun Card(
    imageId: Identifier,
    title: String,
    active: Boolean,
    onActiveChange: (Boolean) -> Unit,
    modifier: Modifier = Modifier,
    description: String? = null,
    locked: Boolean = false,
    lockText: String? = null
) {
    SimpleCard(
        padding = PaddingValues(16.dp),
        active = active,
        onClick = if (!locked) ({ onActiveChange(!active) }) else null,
        modifier = modifier.alpha(if (locked) 0.5f else 1f),
        overlay = {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(16.dp)
            ) {
                if (locked) {
                    SvgIcon(LOCK_ICON, 24.dp, Color.White, Modifier.align(Alignment.TopEnd))
                    if (lockText != null) {
                        Text(
                            text = lockText,
                            style = StudioTypography.light(11),
                            color = StudioColors.Zinc400,
                            modifier = Modifier.align(Alignment.BottomEnd)
                        )
                    }
                } else if (active) {
                    SvgIcon(CHECK_ICON, 24.dp, Color.White, Modifier.align(Alignment.TopEnd))
                }
            }
        }
    ) {
        Column(
            verticalArrangement = Arrangement.SpaceBetween,
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier
                .fillMaxWidth()
                .fillMaxHeight()
        ) {
            Column(
                verticalArrangement = Arrangement.spacedBy(4.dp),
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp)
            ) {
                Text(
                    text = title,
                    style = StudioTypography.semiBold(16),
                    color = StudioColors.Zinc100
                )
                if (description != null) {
                    Text(
                        text = description,
                        style = StudioTypography.regular(13),
                        color = StudioColors.Zinc400
                    )
                }
            }

            Box(
                contentAlignment = Alignment.Center,
                modifier = Modifier
                    .padding(bottom = 32.dp)
                    .size(64.dp)
            ) {
                ResourceImageIcon(
                    location = imageId,
                    size = 64.dp,
                    modifier = Modifier.size(64.dp)
                )
            }
        }
    }
}
